core/runtime/src/main/java/io/quarkiverse/langchain4j/guardrails/OutputGuardrail.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkiverse.langchain4j.guardrails;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.rag.AugmentationResult;
+
+public interface OutputGuardrail {
```

Comment:
Would it make sense to have `OutputGuardrail` implement some core interface like `OutputValidator`, and similarly for the input ? That way different types of input validators can be registered (thinking about for ex simple custom SQL query validators etc)

Reply:
It will not be possible due to the signature difference (including exception as input guard rail won't retry or reprompt) and potential signature conflict. Because guardrails are beans, move the logic in a bean and use it from both beans.